store/src/java/com/zimbra/cs/service/admin/Auth.java

```diff
@@ -74,346 +57,100 @@
+            
+            if (password != null) {
+                ZimbraLog.account.debug("Authenticating account using password.");
+                prov.authAccount(acct, password, AuthContext.Protocol.soap, authCtxt);
+                context.put(Provisioning.AUTH_MODE_KEY, AuthMode.PASSWORD);
+            }
 
-                if (by == AccountBy.name && value.indexOf("@") == -1) {
-                    // first try to get by adminName, which resolves the account under cn=admins,cn=zimbra
-                    // and does not need a domain
-                    acct = prov.get(AccountBy.adminName, value, zsc.getAuthToken());
+            if (AccountUtil.isTwoFactorAccount(acct)) {
+                String twoFactorCode = request.getAttribute(AccountConstants.E_TWO_FACTOR_CODE, null);
+                Element authTokenEl = request.getOptionalElement(AccountConstants.E_AUTH_TOKEN);
+                String reqTokenType = request.getAttribute(AccountConstants.A_TOKEN_TYPE, "");
+                TokenType tokenType = TokenType.fromCode(reqTokenType);
 
-                    // not found, try applying virtual host name
-                    if (acct == null) {
-                        if (virtualHost != null) {
-                            Domain d = prov.get(Key.DomainBy.virtualHostname, virtualHost);
-                            if (d != null)
-                                value = value + "@" + d.getName();
-                        }
-                    }
+                if (authTokenEl == null && StringUtil.isNullOrEmpty(password)) {
+                    throw ServiceException.INVALID_REQUEST("missing required attribute: " + AccountConstants.E_PASSWORD, null);
                 }
 
-                if (acct == null)
-                    acct = prov.get(by, value);
-
-                if (acct == null)
-                    throw AuthFailedServiceException.AUTH_FAILED(value, valuePassedIn, "account not found");
-
-                checkAdmin(acct);
-
-                AccountUtil.addAccountToLogContext(prov, acct.getId(), ZimbraLog.C_NAME, ZimbraLog.C_ID, null);
-
-                ZimbraLog.security.info(ZimbraLog.encodeAttrs(
-                        new String[] {"cmd", "AdminAuth","account", value}));
-
-                Map<String, Object> authCtxt = new HashMap<String, Object>();
-                authCtxt.put(AuthContext.AC_ORIGINATING_CLIENT_IP, context.get(SoapEngine.ORIG_REQUEST_IP));
-                authCtxt.put(AuthContext.AC_REMOTE_IP, context.get(SoapEngine.SOAP_REQUEST_IP));
-                authCtxt.put(AuthContext.AC_ACCOUNT_NAME_PASSEDIN, valuePassedIn);
-                authCtxt.put(AuthContext.AC_USER_AGENT, zsc.getUserAgent());
-                authCtxt.put(AuthContext.AC_AS_ADMIN, Boolean.TRUE);          
+                if (password != null && twoFactorCode == null) {
+                 // Will return a temporary Auth token that will be sent again with two factor code by client
```

Comment:
indentation is not right and also make the commenting consistent to start with small case letter